assets/js/components/KeyMetrics/ChipTabGroup/index.js

```diff
@@ -115,9 +128,16 @@ export default function ChipTabGroup( { allMetricItems, savedItemSlugs } ) {
 		[ hasGeneratingLeadsGroup, hasSellingProductsGroup ]
 	);
 
+	const dynamicGroups = useMemo( () => {
```

Comment:
What is the point of using `useMemo` in this component? This seems to be completely useless, let's remove all `useMemo` from this component if there is no need for them.

Reply:
@eugene-manuilov I am not entirely sure why useMemo was originally used, but I got warnings that I had to wrap mine in useMemo as a result. I have refactored the code somewhat, but we can not get away from needing to assign `allGroups` assignment in useMemo due to it's use as a dep within the useCallback function.

Reply:
This has been addressed in recent commits, thanks @eugene-manuilov

Reply:
Oh.. i see now. Thanks. In this case, it makes sense to use `useMemo`.  Let's add it back for dynamicGroups and keyMetricsGroups.

Reply:
@eugene-manuilov I have added the useMemo back in.